File: backend/app/routers/checkin.py
from datetime import datetime

from fastapi import APIRouter

from app.schemas import ConfirmRequest, ConfirmResponse, ScanResponse

router = APIRouter(prefix="/api/checkin", tags=["checkin"])


@router.post("/scan", response_model=ScanResponse)
def scan() -> dict:
    return {"success": True}


@router.post("/confirm", response_model=ConfirmResponse)
def confirm(payload: ConfirmRequest) -> dict:
    return {"keyNumber": payload.lockerNumber, "timestamp": datetime.now().strftime("%H:%M")}

File: frontend/lib/api/checkin.ts
import { apiRequest } from "./client";
import type { ConfirmRequest, ConfirmResponse, ScanResponse } from "./types";

export function scan(): Promise<ScanResponse> {
  return apiRequest<ScanResponse>("/api/checkin/scan", { method: "POST" });
}

export function confirm(payload: ConfirmRequest): Promise<ConfirmResponse> {
  return apiRequest<ConfirmResponse>("/api/checkin/confirm", { method: "POST", body: payload });
}
